components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityGovernanceServiceImpl.java

```diff
@@ -233,10 +239,34 @@ public ConnectorConfig getConnectorWithConfigs(String tenantDomain,
 
         for (ConnectorConfig connectorConfig : connectorListWithConfigs) {
             if (connectorConfig.getName().equals(connectorName)) {
+                // Should remove this logic eventually.
+                migratePropertyUseNumericToUseAlphaNumeric(connectorName, connectorConfig);
                 return connectorConfig;
             }
         }
         return null;
     }
 
+    /**
+     * This method is used to migrate the property value useNumericCharacters to useAlphanumericCharacters.
+     *
+     * @param connectorName   Name of the connector.
+     * @param connectorConfig Connector configuration.
+     */
+    private void migratePropertyUseNumericToUseAlphaNumeric(String connectorName, ConnectorConfig connectorConfig) {
+
+        if (connectorName.equals("email-otp-authenticator")) {
+            if (connectorConfig.getProperties().length == 6) {
+                boolean isMigrated = Boolean.parseBoolean(connectorConfig.getProperties()[5].getValue());
+                if  (!isMigrated && connectorConfig.getProperties()[4].getValue() != null ) {
+                    // Extract the value of the useNumericCharacters property.
+                    boolean useAlphanumericChars = !Boolean.parseBoolean(connectorConfig.getProperties()[4].getValue());
+                    // Assign the value to the alphanumeric property.
+                    connectorConfig.getProperties()[3].setValue(String.valueOf(useAlphanumericChars));
+                    connectorConfig.getProperties()[5].setValue(String.valueOf(true));
```

Comment:
let's use keys

Reply:
Cannot retrieve values using keys, as `connectorConfig.getProperties()` returns a list of property values (`Property[]`).